### DISCUSSION TITLE
Enable `exactOptionalPropertyTypes` & `noUncheckedIndexedAccess`

### DISCUSSION THREAD

**Discussion by fubhy** [wevm/viem#352] [Idea / Feature Request] [answered]
Following https://github.com/wagmi-dev/viem/commit/8e1b712efc256c16a98408d10557d5f55d8927d7 I think we should enable `exactOptionalPropertyTypes` & `noUncheckedIndexedAccess`.

Enabling both and then browsing the code base for errors highlighted by `pnpm typecheck` reveals a few spots that could benefit from some hardening & runtime checks.

**Comment by jxom** [comment, marked as answer]
Makes sense!
